src/ContentPage/index.js

```diff
@@ -84,12 +91,21 @@ const FAQRoutes = (
     </Switch>
   </MemoryRouter>
 );
-const ContentPage = () => {
+
+type Props = {
+  history: RouterHistory,
+};
+const ContentPage = (props: Props) => {
+  const returnToLogin = () =>
```

Comment:
this should be moved down to `QueryRenderer` as well, no need to pass it down as prop

Reply:
The problem I faced here and why I ended up with this rather ugly solution is that QueryRenderer is instantiated by many different components, and some of the components are not rendered under the "main" MemoryRouter. But I am only interested in the returnToLogin that pushes history to the "main" router(the parent of ContentPage, otherwise I would only redirect to login half of the page, ie. a sub-router like the one of BookingRoutes in Line 48 here. Hence I cannot decorate QueryRenderer with withRouter cause that would not be linked to the desired main MemoryRouter instance.

All of this will be thankfully obsolete with your new proposal for Routing. So I am totally fine having you merge the new routing system first, and then rebasing this PR, however painful. slightly_smiling_face

Reply:
ok, le's merge it and let's suffer in my PR